File: Backend/SistemaAlmacen.Api/Models/Material.cs
namespace SistemaAlmacen.Api.Models;

// Representa un material registrado en el almacén.
public sealed class Material
{
    public int IdMaterial { get; set; }

    public string NumeroParteMaterial { get; set; } =
        string.Empty;

    public string Descripcion { get; set; } =
        string.Empty;

    public string? UnidadMedida { get; set; }

    public string? CodigoBarras { get; set; }

    public string? SerialKits { get; set; }

    public string GenericCode { get; set; } =
        string.Empty;

    // Indica cómo se entrega físicamente el material.
    public string? TipoEmpaque { get; set; }

    // Indica cuántas unidades contiene cada empaque.
    public decimal? StdPack { get; set; }

    public bool Activo { get; set; }
}

File: Frontend/sistema-almacen-web/src/types/material.ts
// Modelo utilizado por otras áreas antiguas del sistema.
export interface Material {
  id: number;
  idMaterial: number;
  materialNumber: string;
  descripcion: string;
  ubicacion: string;
  stock: number;
  stockMinimo: number;
  codigoBarras: string;
  activo: boolean;
}

// Material devuelto por /api/materiales.
export interface MaterialCatalogo {
  idMaterial: number;
  numeroParteMaterial: string;
  descripcion: string;
  unidadMedida: string | null;
  codigoBarras: string | null;
  serialKits: string | null;
  genericCode: string;
  tipoEmpaque: string | null;
  stdPack: number | null;
  activo: boolean;
}

// Datos enviados para registrar un material.
export interface CrearMaterialRequest {
  numeroParteMaterial: string;
  descripcion: string;
  unidadMedida: string | null;
  codigoBarras: string | null;
  serialKits: string | null;
  genericCode: string;
  tipoEmpaque: string | null;
  stdPack: number | null;
}

// Datos enviados para actualizar un material.
export interface ActualizarMaterialRequest
  extends CrearMaterialRequest {
  activo: boolean;
}